### PR TITLE
AMP-28342: Fix team filter for sync endpoint

### PR DESCRIPTION
**Pull request by vchihai**

### CODE REVIEW

File: amp/WEB-INF/src/org/dgfoundation/amp/ar/AmpARFilter.java [hunk 1]
```diff
@@ -878,19 +880,19 @@ public void fillWithDefaultsFilter(Long ampReportId)
...
-        if (request != null)
-            tm = (TeamMember) request.getSession().getAttribute(Constants.CURRENT_MEMBER);
-        
-        if (tm == null || tm.getTeamId() == null )
-            tm  = null;
 
-        if (tm != null) {
+        if (teamMember == null && request != null) {
+            teamMember = (TeamMember) request.getSession().getAttribute(Constants.CURRENT_MEMBER);
+        }
```
Comment (nmandrescu) [inline amp/WEB-INF/src/org/dgfoundation/amp/ar/AmpARFilter.java]: I am wondering if `teamMember` wasn't stored within `AmpARFilter` on purpose. E.g. if user logs out and refreshes a report, will it work ok? Or could there be any other "cached" filter issue?

Comment (vchihai) [reply amp/WEB-INF/src/org/dgfoundation/amp/ar/AmpARFilter.java]: AmpARFilter object is stored in the user session. If the user logs out, the object will not be available anymore.

Comment (nmandrescu) [reply amp/WEB-INF/src/org/dgfoundation/amp/ar/AmpARFilter.java]: will it update correctly when switching workspace?

Comment (vchihai) [reply amp/WEB-INF/src/org/dgfoundation/amp/ar/AmpARFilter.java]: Yes, you are right. It is recreated every time when the user switches to a  new workspace

**Pull request merged**